### ISSUE TITLE
Bug: `capitalized-comments` yelling about an inline comment

### ISSUE THREAD

**Issue by hbergren** [eslint/eslint#13229]
**Tell us about your environment**

```
Environment Info:

Node version: v14.0.0
npm version: v6.14.4
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found
```

**What parser (default, Babel-ESLint, etc.) are you using?**
`@typescript-eslint/parser`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,

  parser: '@typescript-eslint/parser', // Make ESLint compatible with TypeScript
  parserOptions: {
    // Enable linting rules with type information from our tsconfig
    tsconfigRootDir: __dirname,
    project: ['./tsconfig.eslint.json'],

    sourceType: 'module', // Allow the use of imports / ES modules

    ecmaFeatures: {
      impliedStrict: true, // Enable global strict mode
    },
  },

  // Specify global variables that are predefined
  env: {
    node: true, // Enable node global variables & Node.js scoping
    es2020: true, // Add all ECMAScript 2020 globals and automatically set the ecmaVersion parser option to ES2020
  },

  plugins: [
  ],

  extends: [
    '@xpring-eng/eslint-config-base',
  ],
  rules: {},
  overrides: [],
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const example = {
  nonce: 'abc', // a numeric string
}
```

```bash
eslint . --ext .ts
```

**What did you expect to happen?**
I expected to not get a warning on an uncapitalized inline comment.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  55:7   warning  Variable 'example' is declared but its value is never read. Allowed unused names must match /^_/  @typescript-eslint/no-unused-vars-experimental
  55:7   warning  'example' is assigned a value but never used. Allowed unused vars must match /^_/u                @typescript-eslint/no-unused-vars
  56:17  warning  Comments should not begin with a lowercase character                                              capitalized-comments
```

**Are you willing to submit a pull request to fix this bug?**

Yes, with some guidance.

**Comment by anikethsaha** [comment]
This is the intendent behavior if you are using this rule without any option. 

however, I tried with [`ignoreInlineComments`](https://eslint.org/docs/rules/capitalized-comments#ignoreinlinecomments) [here](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNhcGl0YWxpemVkLWNvbW1lbnRzOiBbXCJlcnJvclwiLCBcImFsd2F5c1wiLCB7IFwiaWdub3JlSW5saW5lQ29tbWVudHNcIjogdHJ1ZSB9XSAqL1xuY29uc3QgZXhhbXBsZSA9IHtcbiAgbm9uY2U6ICdhYmMnLCAvLyBhIG51bWVyaWMgc3RyaW5nXG59XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNhcGl0YWxpemVkLWNvbW1lbnRzIjoyfSwiZW52Ijp7fX19) and with this option set to `true` it still reports it so I guess it is a bug with `ignoreInlineComments`.

**Comment by kaicataldo** [comment]
Closing this issue since the question has been answered.

**Comment by hbergren** [comment]
@kaicataldo , it hasn't been answered. There's a bug with the `ignoreInlineComments` option of the `capitalized-comments` rule.

**Comment by kaicataldo** [comment]
Whoops, sorry about that. Reopening.

**Comment by kaicataldo** [comment]
For what it's worth, this looks like it's working as expected to me. From the [documentation](https://eslint.org/docs/rules/capitalized-comments#ignoreinlinecomments) (emphasis my own):

> Setting the ignoreInlineComments option to true means that comments in the middle of code (with a token on the same line as the beginning of the comment, **and another token on the same line as the end of the comment**) will not be reported by this rule.

Since there can't be tokens on the line after a line comment, line comments will always warn.

```js
/* eslint capitalized-comments: ["error", "always", { "ignoreInlineComments": true }] */

const example = {
 // This next line will error.
  nonce: 'abc', // a numeric string
}

const example2 = {
 // This next line will error.
  nonce: 'abc', /* a numeric string */ 
}

const example3 = {
 // This next line will not error.
  nonce: 'abc' /* a numeric string */,
}
```

See demo [here](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNhcGl0YWxpemVkLWNvbW1lbnRzOiBbXCJlcnJvclwiLCBcImFsd2F5c1wiLCB7IFwiaWdub3JlSW5saW5lQ29tbWVudHNcIjogdHJ1ZSB9XSAqL1xuXG5jb25zdCBleGFtcGxlID0ge1xuICBub25jZTogJ2FiYycsIC8vIGEgbnVtZXJpYyBzdHJpbmdcbn1cblxuY29uc3QgZXhhbXBsZTIgPSB7XG4gIG5vbmNlOiAnYWJjJywgLyogYSBudW1lcmljIHN0cmluZyAqL1xufVxuXG5jb25zdCBleGFtcGxlMyA9IHtcbiAgbm9uY2U6ICdhYmMnIC8qIGEgbnVtZXJpYyBzdHJpbmcgKi8sXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==).

@anikethsaha Can you clarify what you think the bug is?

**Comment by mdjermanovic** [comment]
If the intention is to ignore line comments (those starting with `//`), this rule doesn't have an explicit option for that, but `ignorePattern` can be used as a workaround, e.g.:

```js
/* eslint capitalized-comments: ["error", "always", {
  line: {
    ignorePattern: "."
  }
}] */

// line comment lint-free
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNhcGl0YWxpemVkLWNvbW1lbnRzOiBbXCJlcnJvclwiLCBcImFsd2F5c1wiLCB7XG4gIGxpbmU6IHtcbiAgICBpZ25vcmVQYXR0ZXJuOiBcIi5cIlxuICB9XG59XSAqL1xuXG4vLyBsaW5lIGNvbW1lbnQgbGludC1mcmVlXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Comment by anikethsaha** [comment]
> Setting the ignoreInlineComments option to true means that comments in the middle of code (with a token on the same line as the beginning of the comment, **and another token on the same line as the end of the comment**) will not be reported by this rule.

According to this , this seems intentional behavior. 👍 

But I would consider an inline comment to be ignored with this option even if its next to the line 
like this


```js
const example = {
  nonce: 'abc', // a numeric string <---
}

const example2 = {
  nonce: 'abc', /* a numeric string */  // <---
}
```

because `no-inline-comments` do consider them as [inline comments](https://eslint.org/demo#eyJ0ZXh0IjoiY29uc3QgZXhhbXBsZSA9IHtcbiAvLyBUaGlzIG5leHQgbGluZSB3aWxsIGVycm9yLlxuICBub25jZTogJ2FiYycsIC8vIGEgbnVtZXJpYyBzdHJpbmdcbn1cblxuY29uc3QgZXhhbXBsZTIgPSB7XG4gLy8gVGhpcyBuZXh0IGxpbmUgd2lsbCBlcnJvci5cbiAgbm9uY2U6ICdhYmMnLCAvKiBhIG51bWVyaWMgc3RyaW5nICovIFxufVxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby1pbmxpbmUtY29tbWVudHMiOjJ9LCJlbnYiOnt9fX0=). 

I do think it might be discussed earlier. let me know.

**Comment by kaicataldo** [comment]
> because no-inline-comments do consider them as inline comments.

That's a really good point. We should be trying to be consistent with our definitions within our rules. Given this, I wouldn't be against relaxing the definition of inline comments in `capitalized-comments` to mean "a comment that is not the first token on the line", since it wouldn't be a breaking change. Curious how others feel about this!

**Comment by anikethsaha** [comment]
> I wouldn't be against relaxing the definition of inline comments in `capitalized-comments` to mean "a comment that is not the first token on the line", since it wouldn't be a breaking change. Curious how others feel about this!

sorry if I am not getting, did you mean adding notes in docs or changing the rule ?

**Comment by kaicataldo** [comment]
No worries! This would be changing the behavior of the rule (and then updating the docs!).

**Comment by anikethsaha** [comment]
> No worries! This would be changing the behavior of the rule (and then updating the docs!).

cool, yes I think this will be good change 👍 

May be this is an `enhancement` instead of `bug` cause currently, the rule is working as intended 👍

**Comment by anikethsaha** [comment]
Can this be open ? this issue seems reasonable

**Comment by hbergren** [comment]
Echoing @anikethsaha , can this issue be re-opened?
